### ISSUE TITLE
Cmake release should not compile *_test.cc files

### ISSUE THREAD

**Issue by xkszltl** [facebook/rocksdb#3816]
### Expected behavior

No compile error in cmake Release build.

### Actual behavior

In Release build (i.e. with `NDEBUG`):
rocksdb::SyncPoint is not defined due to `#ifdef NDEBUG`

https://github.com/facebook/rocksdb/blob/72942ad7a40d23ef9b6edf6cf1f53628e37c56b4/util/sync_point.h#L44-L48

But still referenced by files:

https://github.com/facebook/rocksdb/blob/72942ad7a40d23ef9b6edf6cf1f53628e37c56b4/util/auto_roll_logger_test.cc#L294

### Steps to reproduce the behavior

```
mkdir -p build
cd $_
cmake -DCMAKE_BUILD_TYPE=Release <other arguments> ..
```

**Comment by maysamyabandeh** [comment]
*_test.cc files are for tests only and thus not meant to be built in release mode.

**Comment by xkszltl** [comment]
Should we guard the .cc code by ifdef or have more clear setup in build system?

It seems to be the root cause of these kind of issues: #2445 where people are confused by compile errors.

**Comment by ajkr** [comment]
Ideally the build system doesn't try to build test files in release mode. Since you're our main cmake contributor lately would you be interested in trying to change it?

**Comment by xkszltl** [comment]
Ah....I've got a nice experience with rocksdb+cmake on Win so just wanna try it on Linux
But seems it's still far from ready...

Would you mind telling me why not?
Why are we relying on build type rather than a BUILD_TEST switch?
The reason I'm asking is, that doesn't make sense to build twice in our build system if we need a release build with quick test to proof it's not gonna crash

**Comment by sagar0** [comment]
Make is the primary build tool used for building RocksDB on Linux, and I believe not enough effort has been spent on updating and cleaning up CMake files for Linux. On the other hand, CMake is mandatory to generate RocksDB visual studio project files on Windows.

It would be awesome if we can port most of the needed functionality from Makefiles to CMakeLists to work seamlessly on Linux as well.

**Comment by mrambacher** [comment]
I believe this was fixed by #6921
